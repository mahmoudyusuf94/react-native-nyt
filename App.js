import React from 'react';
import {Provider} from 'react-redux';
import HomeScreen from './src/components/HomeScreen';
import createStore from './src/createStore';

const store = createStore();

export default () => (
  <Provider store = {store}>
    <HomeScreen />
  </Provider>
)