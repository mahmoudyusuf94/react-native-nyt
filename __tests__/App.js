/**
 * @format
 * @lint-ignore-every XPLATJSCOPYRIGHT1
 */

import 'react-native';
import React from 'react';
import App from '../App';
import configureStore from 'redux-mock-store';
import promiseMiddleware from 'redux-promise';
import { createLogger } from 'redux-logger';
import toJson from 'enzyme-to-json';

import * as enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
enzyme.configure({ adapter: new Adapter() });

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

const logger = createLogger();
const middlewares = [logger, promiseMiddleware]; // you can mock any middlewares here if necessary
const mockStore = configureStore(middlewares);

const initialState = {
    news: [],
    searchTerm: '',
    bookmarks: []
}
it('renders correctly', () => {
    const wrapper = enzyme.shallow(
        <App />,
        {context: {store: mockStore(initialState)}}
    )
    expect(toJson(wrapper.dive())).toMatchSnapshot();
});
