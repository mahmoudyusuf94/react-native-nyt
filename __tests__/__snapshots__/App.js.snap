// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "storeState": Object {
        "bookmarks": Array [],
        "news": Array [],
        "searchTerm": "",
      },
    }
  }
>
  <NavigationContainer />
</ContextProvider>
`;
